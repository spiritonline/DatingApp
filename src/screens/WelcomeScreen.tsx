import { useColorScheme } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { AuthNavigationProp } from '../navigation/types';
import styled from 'styled-components/native';
import { SafeAreaView } from 'react-native-safe-area-context';

export default function WelcomeScreen() {
  const colorScheme = useColorScheme();
  const navigation = useNavigation<AuthNavigationProp>();
  const isDark = colorScheme === 'dark';

  const handleSignUp = () => {
    navigation.navigate('SignUp');
  };

  const handleLogIn = () => {
    navigation.navigate('LogIn');
  };

  return (
    <Container isDark={isDark}>
      <Logo accessibilityRole="image" accessibilityLabel="Dating App Logo">
        Dating App
      </Logo>
      <Tagline accessibilityRole="text">Find your perfect match</Tagline>
      <ButtonsContainer>
        <PrimaryButton 
          onPress={handleSignUp} 
          accessibilityRole="button" 
          accessibilityLabel="Sign Up"
          accessibilityHint="Navigate to sign up screen"
        >
          <ButtonText isDark={false}>Sign Up</ButtonText>
        </PrimaryButton>
        <SecondaryButton 
          onPress={handleLogIn} 
          accessibilityRole="button" 
          accessibilityLabel="Log In"
          accessibilityHint="Navigate to log in screen"
        >
          <ButtonText isDark={isDark}>Log In</ButtonText>
        </SecondaryButton>
      </ButtonsContainer>
    </Container>
  );
}

import { ThemeProps } from '../utils/styled-components';

const Container = styled(SafeAreaView)<ThemeProps>`
  flex: 1;
  justify-content: center;
  align-items: center;
  padding: 20px;
  background-color: ${(props: ThemeProps) => (props.isDark ? '#121212' : '#ffffff')};
`;

const Logo = styled.Text`
  font-size: 36px;
  font-weight: bold;
  color: #FF6B6B;
  margin-bottom: 10px;
`;

const Tagline = styled.Text`
  font-size: 18px;
  color: #666;
  margin-bottom: 60px;
`;

const ButtonsContainer = styled.View`
  width: 100%;
  max-width: 300px;
`;

const BaseButton = styled.TouchableOpacity`
  width: 100%;
  padding: 16px;
  border-radius: 8px;
  align-items: center;
  margin-bottom: 16px;
`;

const PrimaryButton = styled(BaseButton)`
  background-color: #FF6B6B;
`;

const SecondaryButton = styled(BaseButton)<ThemeProps>`
  background-color: transparent;
  border: 1px solid #FF6B6B;
`;

const ButtonText = styled.Text<ThemeProps>`
  font-size: 16px;
  font-weight: bold;
  color: ${(props: ThemeProps) => (props.isDark ? '#FF6B6B' : '#ffffff')};
`;
