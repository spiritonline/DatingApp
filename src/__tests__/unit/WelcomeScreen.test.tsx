import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import WelcomeScreen from '../../screens/WelcomeScreen';

// Mock the navigation
const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockNavigate,
    }),
  };
});

describe('WelcomeScreen', () => {
  beforeEach(() => {
    mockNavigate.mockClear();
  });

  it('renders correctly with sign up and log in buttons', () => {
    const { getByText } = render(
      <NavigationContainer>
        <WelcomeScreen />
      </NavigationContainer>
    );
    
    expect(getByText('Dating App')).toBeTruthy();
    expect(getByText('Find your perfect match')).toBeTruthy();
    expect(getByText('Sign Up')).toBeTruthy();
    expect(getByText('Log In')).toBeTruthy();
  });

  it('navigates to SignUp screen when Sign Up button is pressed', () => {
    const { getByText } = render(
      <NavigationContainer>
        <WelcomeScreen />
      </NavigationContainer>
    );
    
    const signUpButton = getByText('Sign Up');
    fireEvent.press(signUpButton);
    
    expect(mockNavigate).toHaveBeenCalledWith('SignUp');
  });

  it('navigates to LogIn screen when Log In button is pressed', () => {
    const { getByText } = render(
      <NavigationContainer>
        <WelcomeScreen />
      </NavigationContainer>
    );
    
    const logInButton = getByText('Log In');
    fireEvent.press(logInButton);
    
    expect(mockNavigate).toHaveBeenCalledWith('LogIn');
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <WelcomeScreen />
      </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
