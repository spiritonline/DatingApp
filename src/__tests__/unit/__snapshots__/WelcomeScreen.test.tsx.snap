// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  isDark={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    accessibilityLabel="Dating App Logo"
    accessibilityRole="image"
    style={
      {
        "color": "#FF6B6B",
        "fontSize": 36,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Dating App
  </Text>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#666",
        "fontSize": 18,
        "marginBottom": 60,
      }
    }
  >
    Find your perfect match
  </Text>
  <View
    style={
      {
        "maxWidth": 300,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityHint="Navigate to sign up screen"
      accessibilityLabel="Sign Up"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FF6B6B",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginBottom": 16,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        }
      }
    >
      <Text
        isDark={false}
        style={
          {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      accessibilityHint="Navigate to log in screen"
      accessibilityLabel="Log In"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#FF6B6B",
          "borderStyle": "solid",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "marginBottom": 16,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        }
      }
    >
      <Text
        isDark={false}
        style={
          {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Log In
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
